common/src/main/java/bisq/common/persistence/db/NoHeapDB.java

```diff
@@ -0,0 +1,380 @@
+        NoHeapDB db = new NoHeapDB();
+       /* db.createStore(
+                "MyTestDataStore",
+                DataStore.Storage.IN_MEMORY, //or DataStore.Storage.PERSISTED
+                256); // in MB*/
+
+        db.createStore(
+                "MyTestDataStore",
+                DataStore.Storage.PERSISTED, //or DataStore.Storage.PERSISTED
+                50, 5); // in MB
+
+        DataStore store = db.getStore("MyTestDataStore");
+        String sep = "_";
+        String result = store.getString(2 + sep + 100);
+        log.error("result {}", result);
+        String name = "MyTestDataStore";
+
+        byte[] array = new byte[10000000];
+        new Random().nextBytes(array);
+        String random = new String(array, StandardCharsets.UTF_8);
+        log.error(Utilities.readableFileSize(array.length));
+        log.error(Utilities.readableFileSize(random.getBytes(StandardCharsets.UTF_8).length));
+
+        c = store.getRecordCount();
+        key = c + sep;
+        ts = System.currentTimeMillis();
+        String res1 = store.getString(key);
+        // log.error("read took {} ms.  {}", System.currentTimeMillis() - ts, res1);
+        Timer timer = UserThread.runPeriodically(() -> {
+
+            ts = System.currentTimeMillis();
+            key = c + sep;
+            String val1 = random + c;
+            store.putString(key, val1);
+            //log.error("write took {} ms", System.currentTimeMillis() - ts);
+
+            ts = System.currentTimeMillis();
+            String res = store.getString(key);
+            // log.error("read took {} ms. res={}, val1={}, match {}", System.currentTimeMillis() - ts, res, val1, res.equals(val1));
+            //  log.error("read took {} ms. match {}", System.currentTimeMillis() - ts, res.equals(val1));
+            c++;
+            log.error("getFilesize {} getRecordCount {}", Utilities.readableFileSize(store.getFilesize()), store.getRecordCount());
+            System.gc();
+            if (store.getFilesize() > 1800000000) {
+                log.error("too large");
+                System.exit(0);
+            }
+            // 400 000
+          /*  long ts = System.currentTimeMillis();
+            int size = 10000000;
+            for (int i = 0; i < size; i++) {
+                String val = String.valueOf(c * i);
+                String key = c + sep + i;
+                store.putString(key, val); //400 000
+               // log.error("write key/val {}/{}", key, val);
+            }
+
+            log.error("write took {} ms", System.currentTimeMillis() - ts);
+            ts = System.currentTimeMillis();
+            for (int i = 0; i < size; i++) {
+                String key = c + sep + i;
+                String val = store.getString(key);
+                //log.error("read key/val {}/{}", key, val);
+            }
+            log.error("read took {} ms", System.currentTimeMillis() - ts);
+            c++;*/
```

Comment:
Why is this code part commented out?

Reply:
Ah that whole common/src/main/java/bisq/common/persistence/db/ dir should have been deleted. There is a later commit removing it... was from the MappedByteBuffer experiments.